s4cmb/tod.py

```diff
@@ -772,19 +870,30 @@ def map2tod(self, ch):
 
         ## Compute pointing for detector ch
         ra, dec, pa = self.pointing.offset_detector(azd, eld)
-
+        
+        # Perturbed boresight pointing values
+        ra_perturbed, dec_perturbed, pa_perturbed = self.pointing_perturbed.offset_detector(azd, eld)
+        
         ## Compute pointing matrix
         index_global, index_local = self.get_pixel_indices(ra,dec)
+        
+        # Perturbed boresight pointing values
+        index_global_perturbed, index_local_perturbed = self.get_pixel_indices(ra_perturbed,dec_perturbed)
```

Comment:
this should only be computed if the pointing is perturbed to save time and memory, and simply referenced to the unperturbed index_global/index_local vectors if the pointing is not perturbed